rig-core/src/providers/anthropic/completion.rs

```diff
@@ -157,9 +151,20 @@ impl completion::CompletionModel for CompletionModel {
         &self,
         completion_request: completion::CompletionRequest,
     ) -> Result<completion::CompletionResponse<CompletionResponse>, CompletionError> {
+        // Note: Ideally we'd introduce provider-specific Request models to handle the
+        // specific requirements of each provider. For now, we just manually check while
+        // building the request as a raw JSON document.
+
         let prompt_with_context = completion_request.prompt_with_context();
 
-        let request = json!({
+        // Check if max_tokens is set, required for Anthropic
```

Comment:
I would have expected this to be done in the `ClientBuilder`, but I see that while the API docs right now encourage you to use the builder they don't *require* it.

Since `max_tokens` is required, shouldn't it prevent use of Anthropic without this on the type level though? That is, make `max_tokens` required for `Client::new` and fail `ClientBuilder::build` if it's not provided, or alternatively make `new` non-public so that the builder is the only way.

This would have the compiler guide you in the right direction, rather than getting an error in runtime.

Reply:
This would be ideal, but probably should be done as a further PR since it's something we might want to standardize for all the providers. ATM, Anthropic is the only one w/ a `ClientBuilder`.